### PR TITLE
Update publishing.md

### PR DESCRIPTION
**Pull request by tudi2d**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/publishing.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ PR is merged. This is typically done every Tuesday around noon CET.
 
 ## Next Line Release Process
 
-- PR Checks: Ensure there are no outstanding PRs pending to be merged for this version. If there are any, reach out to maintainers and relevant owners of the affected code reminding them of the deadline for the release.
-- [optional] Lock main branch
+- PR Checks: Notify the teams & ensure there are no outstanding PRs pending to be merged for this version. This should be done in time to ensure a smoothe release day. If there are any, reach out to maintainers and relevant owners of the affected code reminding them of the deadline for the release.
```
Comment (camilaibs) [inline docs/publishing.md:18]:
```suggestion
- PR Checks: Notify the teams & ensure there are no outstanding PRs pending to be merged for this version. This should be done in time to ensure a smooth release day. If there are any, reach out to maintainers and relevant owners of the affected code reminding them of the deadline for the release.
```

File: docs/publishing.md [hunk 2]
```diff
@@ -32,6 +32,22 @@ PR is merged. This is typically done every Tuesday around noon CET.
...
+## Main Line Release Process
+
+Additional steps for the main line release
+
+- [Switch Release Mode](#switching-release-modes) to exit pre-release mode. This can be done at any time after the last Next Line Release.
+- Check [`Version Packages` Pull Request](https://github.com/backstage/backstage/pulls?q=is%3Aopen+is%3Apr+in%3Atitle+%22Version+Packages)
+  - Check for mentions of "major" & "breaking" and if they are expected in the current release
+  - Verify the version we are shipping is correct
+  - Check [`.changeset/pre.json`](https://github.com/backstage/backstage/blob/master/.changeset/pre.json) if the `mode` is set to `exit`. If you encounter `mode: "pre"` it indicates a next line release.
+- Create Release Notes
```
Comment (camilaibs) [inline docs/publishing.md:44]: Could we link the template here?

**Review by camilaibs — APPROVED** [review]
💯

**Comment by tudi2d** [comment]
Waiting for https://github.com/backstage/backstage/pull/20225 to be merged to add a link in the documentation

**Review by Rugvip — APPROVED** [review]

**Pull request merged**